lib/pure/httpclient.nim

```diff
@@ -1219,21 +1219,24 @@ proc downloadFile*(client: HttpClient, url: Uri | string, filename: string) =
   defer:
     client.getBody = true
   let resp = client.get(url)
+  
+  if resp.code.is4xx or resp.code.is5xx:
+    raise newException(HttpRequestError, resp.status)
 
   client.bodyStream = newFileStream(filename, fmWrite)
   if client.bodyStream.isNil:
     fileError("Unable to open file")
```

Comment:
future PR's welcome to improve error messages, eg:
`fileError("Unable to open file")`
=>
`fileError("Unable to open file: '%1'" %  filename)` or similar